internal/plugins/darwin/hostinfo.go

```diff
@@ -247,72 +254,101 @@ func (self *HostinfoPlugin) setCloudMetadata(data *HostInfoData) (err error) {
 
 // hardwareOverview structure keeps the values extracted from osX system_profiler.
 type hardwareOverview struct {
-	ModelName          string
-	ModelIdentifier    string
+	ModelName       string
+	ModelIdentifier string
+	Memory          string
+	HardwareUUID    string
+	processorInfo
+}
+
+// processorInfo structure keeps the values extracted from osX system_profiler
+// for the processor information.
+type processorInfo struct {
 	ProcessorName      string
 	ProcessorSpeed     string
-	Memory             string
 	NumberOfProcessors string
 	TotalNumberOfCores string
-	HardwareUUID       string
 }
 
-func (hip *HostinfoPlugin) getHardwareOverview() (hardwareOverview, error) {
-	result := hardwareOverview{}
-
-	out, err := hip.readDataFromCmd("system_profiler", "SPHardwareDataType")
-	if err != nil {
-		return result, err
+// splitRightSubstring returns the first remaining part from the end of a substring
+// and a given separator, if no substring or separator is found, and empty
+// string is returned.
+// Example: splitRightSubstring("Hello: bye$", "Hello: ", "$") -> "bye"
+func splitRightSubstring(output, substring, separator string) string {
```

Comment:
as this is not related to hostInfo I would move it to `pkg/helpers/helpers.go`. WDYT?

Reply:
Definitely, good catch 👍